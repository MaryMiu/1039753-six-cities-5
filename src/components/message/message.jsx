import React, {PureComponent, createRef} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {getErrorMessage} from "../../store/selectors";


class Message extends PureComponent {
  constructor(props) {
    super(props);
    this.messageRef = createRef();
  }

  render() {
    const {error} = this.props;
    setTimeout(() => {
      this.messageRef.current.remove();
    }, 5000);
    return (
      <div ref={this.messageRef} style={{position: `fixed`, width: `100%`, backgroundColor: `#d20000`, color: `#fff`, textAlign: `center`, zIndex: `100`}}>{error}</div>
    );
  }
}

Message.propTypes = {
  error: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  error: getErrorMessage(state),
});

export {Message};
export default connect(mapStateToProps)(Message);
