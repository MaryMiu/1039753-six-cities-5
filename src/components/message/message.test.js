import React from "react";
import renderer from "react-test-renderer";
import {Message} from "./message";

it(`Should Message render correctly`, () => {
  const tree = renderer
    .create(<Message
      error={`Ошибка`}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

