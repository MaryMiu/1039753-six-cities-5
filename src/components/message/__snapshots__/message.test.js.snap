// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Message render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#d20000",
      "color": "#fff",
      "display": "block",
      "position": "fixed",
      "textAlign": "center",
      "width": "100%",
      "zIndex": "100",
    }
  }
>
  Ошибка
</div>
`;
